### PR TITLE
Improve grain growth model documentation

### LINKED ISSUE #25144: Improve grain growth documentation pages
## Reason
The grain growth documentation pages are fairly old and not especially helpful. I want to improve them to make them more helpful to users

## Design
I plan to follow a similar design to what I used in the new LinearizedGrainGrowthModel documentation addressing #21940 

## Impact
This will involve no code changes, just changes to documentation pages.

### PR DESCRIPTION
**Pull request by tonkmr**
Improves the documentation for the grain growth model. Also updates examples problems to represent recommended practices for running such simulations.

Closes #25144

This PR only changes documentation, example input files, and one test input file (gold file is unchanged).

### CODE REVIEW

**Comment by tonkmr** [comment]
The best place to start with this PR is Grain_Growth_Model.md. Every documentation page linked from there now exists.

**Comment by tonkmr** [comment]
@dschwen, @GiudGiud, and @laagesen this is ready to review.

File: modules/phase_field/doc/content/source/actions/GrainGrowthAction.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+
+- One per variable, with a total of `op_num`
+
+  - [TimeDerivative](/TimeDerivative.md) or [ADTimeDerivative](/ADTimeDerivative.md)
+  - [ACGrGrPoly](/ACGrGrPoly.md) or [ADGrainGrowth](/ADGrainGrowth.md)
+  - [ACInterface](/ACInterface.md) or [ADACInterface](/ADACInterface.md)
+
++AuxKernels+
+
+- [BndsCalcAux](/BndsCalcAux.md)
```
Comment (GiudGiud) [inline modules/phase_field/doc/content/source/actions/GrainGrowthAction.md:35]:
```suggestion
- [BndsCalcAux](/BndsCalcAux.md) use to set the grain boundaries visualization variable
```

File: modules/phase_field/doc/content/source/actions/GrainGrowthAction.md [hunk 2]
```diff
@@ -1 +1,47 @@
...
+
+When using the action, you can choose to either use the automatic differentiation (AD) versions of the kernels or the versions with hand-coded Jacobians with the parameter `use_automatic_differentiation`. It defaults to 'false'.
+
+The following objects are created:
+
++Variables+
+
+- One per variable, with a total of `op_num`
+
+  - Order parameter (`var_name_base`)
```
Comment (GiudGiud) [inline modules/phase_field/doc/content/source/actions/GrainGrowthAction.md:19]: no clear what you mean here.
do you mean op_num variables for each var_bame_base ?

File: modules/phase_field/doc/content/source/actions/GrainGrowthAction.md [hunk 3]
```diff
@@ -1 +1,47 @@
...
+  - [ACGrGrPoly](/ACGrGrPoly.md) or [ADGrainGrowth](/ADGrainGrowth.md)
+  - [ACInterface](/ACInterface.md) or [ADACInterface](/ADACInterface.md)
+
++AuxKernels+
+
+- [BndsCalcAux](/BndsCalcAux.md)
+
+## Example Input File Syntax
+
+The `GrainGrowthAction` is accessed through the `Module` block, as shown below.
```
Comment (GiudGiud) [inline modules/phase_field/doc/content/source/actions/GrainGrowthAction.md:39]:
```suggestion
The `GrainGrowthAction` is created as shown below.
```
we no longer use the Modules/ syntax

**Review by GiudGiud — COMMENTED** [review]
didn't see much but SMEs should review

File: modules/phase_field/doc/content/source/materials/GBEvolution.md [hunk 4]
```diff
@@ -1 +1,34 @@
...
+
+## Overview
+
+This material calculates the order parameter mobility $L$, the free energy prefactor $\mu$, and the gradient multiplier $\kappa$ for the grain growth model from [!cite](moelans_quantitative_2008), assuming isotropic grain boundary (GB) properties. The parameters are calculated based on the GB energy $\gamma$, the GB mobility $M_{GB}$, and the phase field interfacial width $w_{GB}$ according to
+
+\begin{equation}
+L = \frac{4}{3} \frac{M_{GB}}{w_{GB}}
+\end{equation}
+\begin{equation}
+\mu = \frac{3}{2} \frac{\gamma}{w_{GB}}
```
Comment (laagesen) [inline modules/phase_field/doc/content/source/materials/GBEvolution.md:13]: please see previous comment about value of the prefactor and switching from \gamma to \sigma for interfacial energy

**Review by laagesen — CHANGES_REQUESTED** [review]
Thanks for helping improve the documentation!

File: modules/phase_field/doc/content/modules/phase_field/Grain_Growth_Model.md [hunk 5]
```diff
@@ -102,8 +126,33 @@ Thus, the area at any time $t$ can be calculated with
...
+
+!listing modules/phase_field/test/tests/grain_growth/thumb.i
+
+## Example Input Files
+
+An example 2D input file for a polycrystal simulation using `GrainTracker` is available. It is should be run on at least 4 processors:
+
+!listing modules/phase_field/examples/grain_growth/grain_growth_2D_graintracker.i
+
+An example 3D input file is also available:
```
[the remark this answers is not in the document]
Comment (tonkmr) [reply modules/phase_field/doc/content/modules/phase_field/Grain_Growth_Model.md:152]: It actually runs fine on 1 core, which is why I didn't put a minimum. I can give a minimum of 1, but that seemed odd.

**Comment by tonkmr** [comment]
@laagesen I addressed your comments.

**Review by laagesen — APPROVED** [review]

**Review by GiudGiud — DISMISSED** [review]
Thanks for documenting all this! Users will appreciate

**Review by GiudGiud — APPROVED** [review]
just rebased it

**Pull request merged**